widget/table.go

```diff
@@ -0,0 +1,299 @@
+package widget
+
+import (
+	"image/color"
+	"math"
+
+	"fyne.io/fyne"
+	"fyne.io/fyne/canvas"
+	"fyne.io/fyne/theme"
+)
+
+// Table widget is a grid of items that can be scrolled and a cell selected.
+// It's performance is provided by caching cell templates created with NewCell and re-using them with UpdateCell.
+// The size of the content rows/columns is returned by the DataSize callback.
+type Table struct {
+	BaseWidget
+
+	DataSize       func() (int, int)
+	NewCell        func() fyne.CanvasObject
+	UpdateCell     func(row int, col int, template fyne.CanvasObject)
+	OnCellSelected func(row int, col int)
+
+	cells                       *tableCells
+	scroll                      *ScrollContainer
```

Comment:
Neither List nor Tree keeps a reference to the scroller, only the renderer does. Why is this needed? Most of the usages are `t.t.scroll`

Reply:
It's used to ask the overlay to update when the content of the scroller is tapped.
It's an indirection to avoid the cells trying to get a reference to the render of the parent component.
I can write the indirection in through another means.

Reply:
The functionality for ensuring the tapped cell is visible in the scroller is a nice-to-have, but not something List or Tree do. You could drop this to simplify the code for this first attempt

Reply:
It's not related to is the cell visible - it's related to how do we ensure the indicator updates when it's tapped

Reply:
It's still in - I don't think the code is too hard to follow, but will leave it to others to determine.

Reply:
If the OnTapped is inside of the renderer like it is in list then I don't think this is a problem and the reference to the scroll container could be removed as Stuart suggests.  https://github.com/fyne-io/fyne/blob/a022e44470ad884476176dd7a423e9bd66e78281/widget/list.go#L261

Reply:
I guess I wanted to avoid creating a new widget type that wrapped all the cells - there can be many hundreds of them visible at any time. So a single Tapped handler in the parent made more sense to me...

Reply:
Perhaps someone can remind me why the line in question is a problem?

Reply:
Not a problem per se, just uncommon for widgets to hold references to subcomponents - mostly held by widgetrenderers. This isn't a blocker for me, just wanted to explore alternatives

Reply:
I guess. I think that popups and scrollers are special cases as they often need to be manipulated when the parent object changes. See Select/SelectEntry.